nemo/collections/nlp/modules/common/transformer/transformer_decoders.py

```diff
@@ -180,31 +180,34 @@ def forward(
         decoder_attn_mask = form_attention_mask(decoder_mask, diagonal=self.diagonal)
         encoder_attn_mask = form_attention_mask(encoder_mask)
         memory_states = self._get_memory_states(decoder_states, decoder_mems_list, 0)
-        if return_mems_as_list:
-            cached_mems_list = [memory_states]
-        else:
-            cached_mems_list = memory_states.unsqueeze(0)
+        if return_mems:
```

Comment:
@stevehuang52 switch to a copy in asr collection for this

Reply:
done